### ISSUE TITLE
[FR] Add a "Bypass Proxy Rule" feature

### ISSUE THREAD

**Issue by dmarkdmark** [hiddify/hiddify-app#1559]
### Feature description

This proxy software works great! I've abandoned both Black Cat and Shadowrocket, and now I only use this one. While using it, I noticed that the service mode only allows users to select the mode manually.

![Clip_2024-12-18_15-05-26](https://github.com/user-attachments/assets/1d1253d6-d228-4696-87e4-b3fe99898446)

Now, I’d like to have a feature where I can input a list of URLs to decide whether they should bypass the proxy or use it. This would work like a blacklist or whitelist to control whether the proxy software is used for accessing these URLs.

For example, if I input a URL like www.123456.com, I want to control whether it uses the proxy or not. If I choose to use the proxy, the software will route the request through the proxy to access the website. If I choose not to use the proxy, it will use my direct network connection to access the website instead.

It's not about rewriting the service mode, but adding some customization. Sometimes accessing a website directly (without the proxy) is faster, but in some cases, it's not clear why the proxy makes access very slow. So, I need this feature to control whether a specific website should go through the proxy or not.

**Comment by netrolite** [comment]
I absolutely second this! Came here to make a request about this exact feature and stumbled upon this issue

**Comment by S-Vladislav** [comment]
Agree, something like this will be very useful, also on mobile you already can choose which apps to proxy, which is a cool feature

**Comment by dmarkdmark** [comment]
+1

**Comment by LzxLeo** [comment]
The function is really a must. For example, gstatic cannot be connected directly as I'm in some high-risk area in China. Then, I need force it to connect via proxy.

**Comment by dgutyanghs** [comment]
indeed, this feature will be greate value for Hiddify.
